### DISCUSSION TITLE
How to I use the microtype packlage in Quarto?

### DISCUSSION THREAD

**Discussion by e05bf027** [quarto-dev/quarto-cli#6331] [Q&A] [answered]
### Please help me configure the microtype package in the YAML header

I am writing a thesis, and really find Quarto to be great. I am getting to the point where I am fine-tuning the typesetting. The `microtype` package for Latex is what I am interested in integrating properly. I know that Quarto has an option in the YAML header called `microtypeoptions`. I don't actually know literally what to write here to feed the instructions to the package.

I often find this with the YAML stuff; the documentation on the Quarto site doesn't make clear what I literally need to write to configure the commands correctly. In most cases I can figure it out but this one has stumped me. If anyone could advise I'd me most grateful.

I have tried putting options separated by commas, with and without quotation marks, inside square brackets or not, and have added `/usepackage{microtype}' to the relevant part of the YAML in case it needed it. I am unsure why I can't get this.

**I am running Quarto through Rstudio version 2023.06.0+421 on a M1 Macbook Air, with Tinytex installed**

**Comment by mcanouil** [comment]
Did you read the guide for PDF/LaTeX?

- <https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes>

It's pretty much raw LaTeX, so I suggest you get familiar with the LaTeX code required to configure this particular LaTeX package (see package documentation), then use those commands as des robes in the guide under the "include" section.

**Comment by cderv** [comment]
`microtypeoptions` field in YAML will be passed to LaTeX template. This is a pandoc feature that still apply in Quarto as the default LaTeX template has theses line 
https://github.com/quarto-dev/quarto-cli/blob/d4f620e5cf64e4c1b767943f67825dd531de609a/src/resources/formats/pdf/pandoc/template.tex#L192-L195

It means that if **microtype** is installed on the LaTeX installation, it will be used , and if  `microtypeoptions` is set then it will be passed as options. 

Did you look into your produced `.tex` file ? Is microtype loaded in the file ? Are options passed ? 

I would suggest to look into the .tex file that you produce by rendering Quarto document and check what is inside. 

It is possible that everything is inserted as it should but this is not working as you expect for some reason (LaTeX conflict ?)

**Reply by mcanouil** [reply]
`keep-tex: true` in your YAML front matter to check what’s inside the tex file.

**Comment by mcanouil** [comment, marked as answer]
The `microtypeoptions` is to be defined as follow[^1]:

```yaml
microtypeoptions:
  - protrusion=true
  - verbose=true
```

[^1]: See <https://ctan.tetaneutral.net/macros/latex/contrib/microtype/microtype.pdf>. Also note that all options are not available to all PDF engines, _i.e._, some only works for `pdflatex` for example.

**Reply by e05bf027** [reply]
This is exactly what I was hoping for. Thank you
